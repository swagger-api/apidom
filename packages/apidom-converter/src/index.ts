import { ParseResultElement } from '@swagger-api/apidom-core';
import { mergeOptions, bundle, File } from '@swagger-api/apidom-reference';

import defaultOptions, { ConverterOptions } from './options';
import ConvertError from './errors/ConvertError';
import UnmatchedConvertStrategyError from './errors/UnmatchedConvertStrategyError';

export { ConvertError, UnmatchedConvertStrategyError };

/**
 * `convertApiDOM` already assumes that the ApiDOM is bundled.
 */
export const convertApiDOM = async (element: ParseResultElement, options = {}) => {
  const mergedOptions = mergeOptions(defaultOptions, options || {}) as ConverterOptions;
  const file = File({
    uri: mergedOptions.resolve.baseURI,
    parseResult: element,
    mediaType: mergedOptions.convert.sourceMediaType || mergedOptions.parse.mediaType,
  });
  const strategy = mergedOptions.convert.strategies.find((s) => s.canConvert(file, mergedOptions));

  if (typeof strategy === 'undefined') {
    throw new UnmatchedConvertStrategyError(file.uri);
  }

  return strategy.convert(file, mergedOptions);
};

const convert = async (uri: string, options = {}) => {
  const mergedOptions = mergeOptions(defaultOptions, options || {}) as ConverterOptions;
  const parseResult = await bundle(uri, mergedOptions);

  return convertApiDOM(parseResult, mergedOptions);
};

export default convert;
