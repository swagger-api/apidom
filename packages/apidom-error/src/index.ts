// base error classes
export { default as ApiDOMError } from './ApiDOMError';
export { default as ApiDOMAggregateError } from './ApiDOMAggregateError';
export { default as ApiDOMStructuredError } from './ApiDOMStructuredError';
export type { default as ApiDOMErrorOptions } from './ApiDOMErrorOptions';
// generic custom error classes
export { default as UnsupportedOperationError } from './UnsupportedOperationError';
export { default as NotImplementedError } from './NotImplementedError';
