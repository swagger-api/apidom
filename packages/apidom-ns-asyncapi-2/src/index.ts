import './refractor/registration';

export {
  isRefElement,
  isLinkElement,
  isMemberElement,
  isObjectElement,
  isArrayElement,
  isBooleanElement,
  isNullElement,
  isElement,
  isNumberElement,
  isStringElement,
} from '@swagger-api/apidom-core';

export { default as mediaTypes, AsyncAPIMediaTypes } from './media-types';

// eslint-disable-next-line no-restricted-exports
export { default } from './namespace';

export { default as refractorPluginReplaceEmptyElement } from './refractor/plugins/replace-empty-element';

export {
  isAsyncApi2Element,
  isAsyncApiVersionElement,
  isChannelBindingsElement,
  isChannelItemElement,
  isChannelItemElementExternal,
  isChannelsElement,
  isComponentsElement,
  isContactElement,
  isIdentifierElement,
  isInfoElement,
  isLicenseElement,
  isOperationElement,
  isParameterElement,
  isParametersElement,
  isReferenceElement,
  isReferenceElementExternal,
  isSchemaElement,
  isBooleanJsonSchemaElement,
  isSecurityRequirementElement,
  isServerElement,
  isServerBindingsElement,
  isServersElement,
  isServerVariableElement,
} from './predicates';

export {
  isAsyncApi2LikeElement,
  isParameterLikeElement,
  isReferenceLikeElement,
  isSchemaLikeElement,
  isSecurityRequirementLikeElement,
  isServerLikeElement,
  isAsyncApiExtension,
} from './refractor/predicates';

export { keyMap, getNodeType } from './traversal/visitor';

/**
 * AsyncApi >= 2.0.0 <=2.6.0 specification elements.
 */
export { default as AsyncApi2Element } from './elements/AsyncApi2';
export { default as AsyncApiVersionElement } from './elements/AsyncApiVersion';
export { default as ChannelBindingsElement } from './elements/ChannelBindings';
export { default as ChannelItemElement } from './elements/ChannelItem';
export { default as ChannelsElement } from './elements/Channels';
export { default as ComponentsElement } from './elements/Components';
export { default as ContactElement } from './elements/Contact';
export { default as CorrelationIDElement } from './elements/CorrelationID';
export { default as DefaultContentTypeElement } from './elements/DefaultContentType';
export { default as ExternalDocumentationElement } from './elements/ExternalDocumentation';
export { default as IdentifierElement } from './elements/Identifier';
export { default as InfoElement } from './elements/Info';
export { default as LicenseElement } from './elements/License';
export { default as MessageElement } from './elements/Message';
export { default as MessageBindingsElement } from './elements/MessageBindings';
export { default as MessageExampleElement } from './elements/MessageExample';
export { default as MessageTraitElement } from './elements/MessageTrait';
export { default as OAuthFlowElement } from './elements/OAuthFlow';
export { default as OAuthFlowsElement } from './elements/OAuthFlows';
export { default as OperationElement } from './elements/Operation';
export { default as OperationBindingsElement } from './elements/OperationBindings';
export { default as OperationTraitElement } from './elements/OperationTrait';
export { default as ParameterElement } from './elements/Parameter';
export { default as ParametersElement } from './elements/Parameters';
export { default as ReferenceElement } from './elements/Reference';
export { default as SchemaElement } from './elements/Schema';
export { default as SecurityRequirementElement } from './elements/SecurityRequirement';
export { default as SecuritySchemeElement } from './elements/SecurityScheme';
export { default as ServerElement } from './elements/Server';
export { default as ServerBindingsElement } from './elements/ServerBindings';
export { default as ServersElement } from './elements/Servers';
export { default as ServerVariableElement } from './elements/ServerVariable';
export { default as TagElement } from './elements/Tag';
export { default as TagsElement } from './elements/Tags';

/**
 * Binding elements.
 */
// AMQP 0-9-1
export { default as AmqpChannelBindingElement } from './elements/bindings/amqp/AmqpChannelBinding';
export { default as AmqpMessageBindingElement } from './elements/bindings/amqp/AmqpMessageBinding';
export { default as AmqpOperationBindingElement } from './elements/bindings/amqp/AmqpOperationBinding';
export { default as AmqpServerBindingElement } from './elements/bindings/amqp/AmqpServerBinding';
// AMQP 1.0
export { default as Amqp1ChannelBindingElement } from './elements/bindings/amqp1/Amqp1ChannelBinding';
export { default as Amqp1MessageBindingElement } from './elements/bindings/amqp1/Amqp1MessageBinding';
export { default as Amqp1OperationBindingElement } from './elements/bindings/amqp1/Amqp1OperationBinding';
export { default as Amqp1ServerBindingElement } from './elements/bindings/amqp1/Amqp1ServerBinding';
// Anypoint MQ
export { default as AnypointmqChannelBindingElement } from './elements/bindings/anypointmq/AnypointmqChannelBinding';
export { default as AnypointmqMessageBindingElement } from './elements/bindings/anypointmq/AnypointmqMessageBinding';
export { default as AnypointmqOperationBindingElement } from './elements/bindings/anypointmq/AnypointmqOperationBinding';
export { default as AnypointmqServerBindingElement } from './elements/bindings/anypointmq/AnypointmqServerBinding';
// Google Cloud Pub/Sub
export { default as GooglepubsubChannelBindingElement } from './elements/bindings/googlepubsub/GooglepubsubChannelBinding';
export { default as GooglepubsubMessageBindingElement } from './elements/bindings/googlepubsub/GooglepubsubMessageBinding';
export { default as GooglepubsubOperationBindingElement } from './elements/bindings/googlepubsub/GooglepubsubOperationBinding';
export { default as GooglepubsubServerBindingElement } from './elements/bindings/googlepubsub/GooglepubsubServerBinding';

// HTTP
export { default as HttpChannelBindingElement } from './elements/bindings/http/HttpChannelBinding';
export { default as HttpMessageBindingElement } from './elements/bindings/http/HttpMessageBinding';
export { default as HttpOperationBindingElement } from './elements/bindings/http/HttpOperationBinding';
export { default as HttpServerBindingElement } from './elements/bindings/http/HttpServerBinding';
// IBM MQ
export { default as IbmmqChannelBindingElement } from './elements/bindings/ibmmq/IbmmqChannelBinding';
export { default as IbmmqMessageBindingElement } from './elements/bindings/ibmmq/IbmmqMessageBinding';
export { default as IbmmqOperationBindingElement } from './elements/bindings/ibmmq/IbmmqOperationBinding';
export { default as IbmmqServerBindingElement } from './elements/bindings/ibmmq/IbmmqServerBinding';
// JMS
export { default as JmsChannelBindingElement } from './elements/bindings/jms/JmsChannelBinding';
export { default as JmsMessageBindingElement } from './elements/bindings/jms/JmsMessageBinding';
export { default as JmsOperationBindingElement } from './elements/bindings/jms/JmsOperationBinding';
export { default as JmsServerBindingElement } from './elements/bindings/jms/JmsServerBinding';
// Kafka
export { default as KafkaChannelBindingElement } from './elements/bindings/kafka/KafkaChannelBinding';
export { default as KafkaMessageBindingElement } from './elements/bindings/kafka/KafkaMessageBinding';
export { default as KafkaOperationBindingElement } from './elements/bindings/kafka/KafkaOperationBinding';
export { default as KafkaServerBindingElement } from './elements/bindings/kafka/KafkaServerBinding';
// Mercure
export { default as MercureChannelBindingElement } from './elements/bindings/mercure/MercureChannelBinding';
export { default as MercureMessageBindingElement } from './elements/bindings/mercure/MercureMessageBinding';
export { default as MercureOperationBindingElement } from './elements/bindings/mercure/MercureOperationBinding';
export { default as MercureServerBindingElement } from './elements/bindings/mercure/MercureServerBinding';
// MQTT
export { default as MqttChannelBindingElement } from './elements/bindings/mqtt/MqttChannelBinding';
export { default as MqttMessageBindingElement } from './elements/bindings/mqtt/MqttMessageBinding';
export { default as MqttOperationBindingElement } from './elements/bindings/mqtt/MqttOperationBinding';
export { default as MqttServerBindingElement } from './elements/bindings/mqtt/MqttServerBinding';
// MQTT 5
export { default as Mqtt5ChannelBindingElement } from './elements/bindings/mqtt5/Mqtt5ChannelBinding';
export { default as Mqtt5MessageBindingElement } from './elements/bindings/mqtt5/Mqtt5MessageBinding';
export { default as Mqtt5OperationBindingElement } from './elements/bindings/mqtt5/Mqtt5OperationBinding';
export { default as Mqtt5ServerBindingElement } from './elements/bindings/mqtt5/Mqtt5ServerBinding';
// NATS
export { default as NatsChannelBindingElement } from './elements/bindings/nats/NatsChannelBinding';
export { default as NatsMessageBindingElement } from './elements/bindings/nats/NatsMessageBinding';
export { default as NatsOperationBindingElement } from './elements/bindings/nats/NatsOperationBinding';
export { default as NatsServerBindingElement } from './elements/bindings/nats/NatsServerBinding';
// Pulsar
export { default as PulsarChannelBindingElement } from './elements/bindings/pulsar/PulsarChannelBinding';
export { default as PulsarMessageBindingElement } from './elements/bindings/pulsar/PulsarMessageBinding';
export { default as PulsarOperationBindingElement } from './elements/bindings/pulsar/PulsarOperationBinding';
export { default as PulsarServerBindingElement } from './elements/bindings/pulsar/PulsarServerBinding';
// Redis
export { default as RedisChannelBindingElement } from './elements/bindings/redis/RedisChannelBinding';
export { default as RedisMessageBindingElement } from './elements/bindings/redis/RedisMessageBinding';
export { default as RedisOperationBindingElement } from './elements/bindings/redis/RedisOperationBinding';
export { default as RedisServerBindingElement } from './elements/bindings/redis/RedisServerBinding';
// SNS
export { default as SnsChannelBindingElement } from './elements/bindings/sns/SnsChannelBinding';
export { default as SnsMessageBindingElement } from './elements/bindings/sns/SnsMessageBinding';
export { default as SnsOperationBindingElement } from './elements/bindings/sns/SnsOperationBinding';
export { default as SnsServerBindingElement } from './elements/bindings/sns/SnsServerBinding';
// Solace
export { default as SolaceChannelBindingElement } from './elements/bindings/solace/SolaceChannelBinding';
export { default as SolaceMessageBindingElement } from './elements/bindings/solace/SolaceMessageBinding';
export { default as SolaceOperationBindingElement } from './elements/bindings/solace/SolaceOperationBinding';
export { default as SolaceServerBindingElement } from './elements/bindings/solace/SolaceServerBinding';
// SQS
export { default as SqsChannelBindingElement } from './elements/bindings/sqs/SqsChannelBinding';
export { default as SqsMessageBindingElement } from './elements/bindings/sqs/SqsMessageBinding';
export { default as SqsOperationBindingElement } from './elements/bindings/sqs/SqsOperationBinding';
export { default as SqsServerBindingElement } from './elements/bindings/sqs/SqsServerBinding';
// STOMP
export { default as StompChannelBindingElement } from './elements/bindings/stomp/StompChannelBinding';
export { default as StompMessageBindingElement } from './elements/bindings/stomp/StompMessageBinding';
export { default as StompOperationBindingElement } from './elements/bindings/stomp/StompOperationBinding';
export { default as StompServerBindingElement } from './elements/bindings/stomp/StompServerBinding';
// WebSocket
export { default as WebSocketChannelBindingElement } from './elements/bindings/ws/WebSocketChannelBinding';
export { default as WebSocketMessageBindingElement } from './elements/bindings/ws/WebSocketMessageBinding';
export { default as WebSocketOperationBindingElement } from './elements/bindings/ws/WebSocketOperationBinding';
export { default as WebSocketServerBindingElement } from './elements/bindings/ws/WebSocketServerBinding';
