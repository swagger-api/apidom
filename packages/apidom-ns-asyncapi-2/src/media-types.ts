import { last } from 'ramda';
import { MediaTypes } from '@swagger-api/apidom-core';

type Format = 'generic' | 'json' | 'yaml';

export class AsyncAPIMediaTypes extends MediaTypes<string> {
  filterByFormat(format: Format = 'generic') {
    const effectiveFormat = format === 'generic' ? 'asyncapi;version' : format;
    return this.filter((mediaType) => mediaType.includes(effectiveFormat));
  }

  findBy(version = '2.4.0', format: Format = 'generic') {
    const search =
      format === 'generic'
        ? `vnd.aai.asyncapi;version=${version}`
        : `vnd.aai.asyncapi+${format};version=${version}`;
    const found = this.find((mediaType) => mediaType.includes(search));

    return found || this.unknownMediaType;
  }

  latest(format: Format = 'generic') {
    return last(this.filterByFormat(format)) as string;
  }
}

const mediaTypes = new AsyncAPIMediaTypes(
  'application/vnd.aai.asyncapi;version=2.0.0',
  'application/vnd.aai.asyncapi+json;version=2.0.0',
  'application/vnd.aai.asyncapi+yaml;version=2.0.0',
  'application/vnd.aai.asyncapi;version=2.1.0',
  'application/vnd.aai.asyncapi+json;version=2.1.0',
  'application/vnd.aai.asyncapi+yaml;version=2.1.0',
  'application/vnd.aai.asyncapi;version=2.2.0',
  'application/vnd.aai.asyncapi+json;version=2.2.0',
  'application/vnd.aai.asyncapi+yaml;version=2.2.0',
  'application/vnd.aai.asyncapi;version=2.3.0',
  'application/vnd.aai.asyncapi+json;version=2.3.0',
  'application/vnd.aai.asyncapi+yaml;version=2.3.0',
  'application/vnd.aai.asyncapi;version=2.4.0',
  'application/vnd.aai.asyncapi+json;version=2.4.0',
  'application/vnd.aai.asyncapi+yaml;version=2.4.0',
);

export default mediaTypes;
