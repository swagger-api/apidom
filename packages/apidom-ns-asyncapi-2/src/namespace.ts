import { NamespacePluginOptions } from '@swagger-api/apidom-core';

/**
 * AsyncApi >= 2.0.0 <=2.5.0 specification elements.
 */
import AsyncApi2Element from './elements/AsyncApi2';
import AsyncApiVersionElement from './elements/AsyncApiVersion';
import ChannelBindingsElement from './elements/ChannelBindings';
import ChannelItemElement from './elements/ChannelItem';
import ChannelsElement from './elements/Channels';
import ComponentsElement from './elements/Components';
import ContactElement from './elements/Contact';
import CorrelationIDElement from './elements/CorrelationID';
import DefaultContentTypeElement from './elements/DefaultContentType';
import ExternalDocumentationElement from './elements/ExternalDocumentation';
import IdentifierElement from './elements/Identifier';
import InfoElement from './elements/Info';
import LicenseElement from './elements/License';
import MessageElement from './elements/Message';
import MessageBindingsElement from './elements/MessageBindings';
import MessageExampleElement from './elements/MessageExample';
import MessageTraitElement from './elements/MessageTrait';
import OAuthFlowElement from './elements/OAuthFlow';
import OAuthFlowsElement from './elements/OAuthFlows';
import OperationElement from './elements/Operation';
import OperationBindingsElement from './elements/OperationBindings';
import OperationTraitElement from './elements/OperationTrait';
import ParameterElement from './elements/Parameter';
import ParametersElement from './elements/Parameters';
import ReferenceElement from './elements/Reference';
import SchemaElement from './elements/Schema';
import SecurityRequirementElement from './elements/SecurityRequirement';
import SecuritySchemeElement from './elements/SecurityScheme';
import ServerElement from './elements/Server';
import ServerBindingsElement from './elements/ServerBindings';
import ServersElement from './elements/Servers';
import ServerVariableElement from './elements/ServerVariable';
import TagElement from './elements/Tag';
import TagsElement from './elements/Tags';
/**
 * Binding elements.
 */
// AMQP 0-9-1
import AmqpChannelBindingElement from './elements/bindings/amqp/AmqpChannelBinding';
import AmqpMessageBindingElement from './elements/bindings/amqp/AmqpMessageBinding';
import AmqpOperationBindingElement from './elements/bindings/amqp/AmqpOperationBinding';
import AmqpServerBindingElement from './elements/bindings/amqp/AmqpServerBinding';
// AMQP 1.0
import Amqp1ChannelBindingElement from './elements/bindings/amqp1/Amqp1ChannelBinding';
import Amqp1MessageBindingElement from './elements/bindings/amqp1/Amqp1MessageBinding';
import Amqp1OperationBindingElement from './elements/bindings/amqp1/Amqp1OperationBinding';
import Amqp1ServerBindingElement from './elements/bindings/amqp1/Amqp1ServerBinding';
// Anypoint MQ
import AnypointmqChannelBindingElement from './elements/bindings/anypointmq/AnypointmqChannelBinding';
import AnypointmqMessageBindingElement from './elements/bindings/anypointmq/AnypointmqMessageBinding';
import AnypointmqOperationBindingElement from './elements/bindings/anypointmq/AnypointmqOperationBinding';
import AnypointmqServerBindingElement from './elements/bindings/anypointmq/AnypointmqServerBinding';
// Google Cloud Pub/Sub
import GooglepubsubChannelBindingElement from './elements/bindings/googlepubsub/GooglepubsubChannelBinding';
import GooglepubsubMessageBindingElement from './elements/bindings/googlepubsub/GooglepubsubMessageBinding';
import GooglepubsubOperationBindingElement from './elements/bindings/googlepubsub/GooglepubsubOperationBinding';
import GooglepubsubServerBindingElement from './elements/bindings/googlepubsub/GooglepubsubServerBinding';
// HTTP
import HttpChannelBindingElement from './elements/bindings/http/HttpChannelBinding';
import HttpMessageBindingElement from './elements/bindings/http/HttpMessageBinding';
import HttpOperationBindingElement from './elements/bindings/http/HttpOperationBinding';
import HttpServerBindingElement from './elements/bindings/http/HttpServerBinding';
// IBM MQ
import IbmmqChannelBindingElement from './elements/bindings/ibmmq/IbmmqChannelBinding';
import IbmmqMessageBindingElement from './elements/bindings/ibmmq/IbmmqMessageBinding';
import IbmmqOperationBindingElement from './elements/bindings/ibmmq/IbmmqOperationBinding';
import IbmmqServerBindingElement from './elements/bindings/ibmmq/IbmmqServerBinding';
// JMS
import JmsChannelBindingElement from './elements/bindings/jms/JmsChannelBinding';
import JmsMessageBindingElement from './elements/bindings/jms/JmsMessageBinding';
import JmsOperationBindingElement from './elements/bindings/jms/JmsOperationBinding';
import JmsServerBindingElement from './elements/bindings/jms/JmsServerBinding';
// Kafka
import KafkaChannelBindingElement from './elements/bindings/kafka/KafkaChannelBinding';
import KafkaMessageBindingElement from './elements/bindings/kafka/KafkaMessageBinding';
import KafkaOperationBindingElement from './elements/bindings/kafka/KafkaOperationBinding';
import KafkaServerBindingElement from './elements/bindings/kafka/KafkaServerBinding';
// Mercure
import MercureChannelBindingElement from './elements/bindings/mercure/MercureChannelBinding';
import MercureMessageBindingElement from './elements/bindings/mercure/MercureMessageBinding';
import MercureOperationBindingElement from './elements/bindings/mercure/MercureOperationBinding';
import MercureServerBindingElement from './elements/bindings/mercure/MercureServerBinding';
// MQTT
import MqttChannelBindingElement from './elements/bindings/mqtt/MqttChannelBinding';
import MqttMessageBindingElement from './elements/bindings/mqtt/MqttMessageBinding';
import MqttOperationBindingElement from './elements/bindings/mqtt/MqttOperationBinding';
import MqttServerBindingElement from './elements/bindings/mqtt/MqttServerBinding';
// MQTT 5
import Mqtt5ChannelBindingElement from './elements/bindings/mqtt5/Mqtt5ChannelBinding';
import Mqtt5MessageBindingElement from './elements/bindings/mqtt5/Mqtt5MessageBinding';
import Mqtt5OperationBindingElement from './elements/bindings/mqtt5/Mqtt5OperationBinding';
import Mqtt5ServerBindingElement from './elements/bindings/mqtt5/Mqtt5ServerBinding';
// NATS
import NatsChannelBindingElement from './elements/bindings/nats/NatsChannelBinding';
import NatsMessageBindingElement from './elements/bindings/nats/NatsMessageBinding';
import NatsOperationBindingElement from './elements/bindings/nats/NatsOperationBinding';
import NatsServerBindingElement from './elements/bindings/nats/NatsServerBinding';
// Redis
import RedisChannelBindingElement from './elements/bindings/redis/RedisChannelBinding';
import RedisMessageBindingElement from './elements/bindings/redis/RedisMessageBinding';
import RedisOperationBindingElement from './elements/bindings/redis/RedisOperationBinding';
import RedisServerBindingElement from './elements/bindings/redis/RedisServerBinding';
// SNS
import SnsChannelBindingElement from './elements/bindings/sns/SnsChannelBinding';
import SnsMessageBindingElement from './elements/bindings/sns/SnsMessageBinding';
import SnsOperationBindingElement from './elements/bindings/sns/SnsOperationBinding';
import SnsServerBindingElement from './elements/bindings/sns/SnsServerBinding';
// Solace
import SolaceChannelBindingElement from './elements/bindings/solace/SolaceChannelBinding';
import SolaceMessageBindingElement from './elements/bindings/solace/SolaceMessageBinding';
import SolaceOperationBindingElement from './elements/bindings/solace/SolaceOperationBinding';
import SolaceServerBindingElement from './elements/bindings/solace/SolaceServerBinding';
// SQS
import SqsChannelBindingElement from './elements/bindings/sqs/SqsChannelBinding';
import SqsMessageBindingElement from './elements/bindings/sqs/SqsMessageBinding';
import SqsOperationBindingElement from './elements/bindings/sqs/SqsOperationBinding';
import SqsServerBindingElement from './elements/bindings/sqs/SqsServerBinding';
// STOMP
import StompChannelBindingElement from './elements/bindings/stomp/StompChannelBinding';
import StompMessageBindingElement from './elements/bindings/stomp/StompMessageBinding';
import StompOperationBindingElement from './elements/bindings/stomp/StompOperationBinding';
import StompServerBindingElement from './elements/bindings/stomp/StompServerBinding';
// WebSocket
import WebSocketChannelBindingElement from './elements/bindings/ws/WebSocketChannelBinding';
import WebSocketMessageBindingElement from './elements/bindings/ws/WebSocketMessageBinding';
import WebSocketOperationBindingElement from './elements/bindings/ws/WebSocketOperationBinding';
import WebSocketServerBindingElement from './elements/bindings/ws/WebSocketServerBinding';

const asyncApi2 = {
  namespace: (options: NamespacePluginOptions) => {
    const { base } = options;

    /**
     * AsyncApi >= 2.0.0 <=2.5.0 specification elements.
     */
    base.register('asyncApi2', AsyncApi2Element);
    base.register('asyncApiVersion', AsyncApiVersionElement);
    base.register('channelBindings', ChannelBindingsElement);
    base.register('channelItem', ChannelItemElement);
    base.register('channels', ChannelsElement);
    base.register('components', ComponentsElement);
    base.register('contact', ContactElement);
    base.register('correlationID', CorrelationIDElement);
    base.register('defaultContentType', DefaultContentTypeElement);
    base.register('externalDocumentation', ExternalDocumentationElement);
    base.register('identifier', IdentifierElement);
    base.register('info', InfoElement);
    base.register('license', LicenseElement);
    base.register('message', MessageElement);
    base.register('messageBindings', MessageBindingsElement);
    base.register('messageExample', MessageExampleElement);
    base.register('messageTrait', MessageTraitElement);
    base.register('oAuthFlow', OAuthFlowElement);
    base.register('oAuthFlows', OAuthFlowsElement);
    base.register('operation', OperationElement);
    base.register('operationBindings', OperationBindingsElement);
    base.register('operationTrait', OperationTraitElement);
    base.register('parameter', ParameterElement);
    base.register('parameters', ParametersElement);
    base.register('reference', ReferenceElement);
    base.register('schema', SchemaElement);
    base.register('securityRequirement', SecurityRequirementElement);
    base.register('securityScheme', SecuritySchemeElement);
    base.register('server', ServerElement);
    base.register('serverBindings', ServerBindingsElement);
    base.register('servers', ServersElement);
    base.register('serverVariable', ServerVariableElement);
    base.register('tag', TagElement);
    base.register('tags', TagsElement);

    /**
     * Binding elements.
     */
    // AMQP 0-9-1
    base.register('amqpChannelBinding', AmqpChannelBindingElement);
    base.register('amqpMessageBinding', AmqpMessageBindingElement);
    base.register('amqpOperationBinding', AmqpOperationBindingElement);
    base.register('amqpServerBinding', AmqpServerBindingElement);
    // AMQP 1.0
    base.register('amqp1ChannelBinding', Amqp1ChannelBindingElement);
    base.register('amqp1MessageBinding', Amqp1MessageBindingElement);
    base.register('amqp1OperationBinding', Amqp1OperationBindingElement);
    base.register('amqp1ServerBinding', Amqp1ServerBindingElement);
    // Anypoint MQ
    base.register('anypointmqChannelBinding', AnypointmqChannelBindingElement);
    base.register('anypointmqMessageBinding', AnypointmqMessageBindingElement);
    base.register('anypointmqOperationBinding', AnypointmqOperationBindingElement);
    base.register('anypointmqServerBinding', AnypointmqServerBindingElement);
    // Google Cloud Pub/Sub
    base.register('googlepubsubChannelBinding', GooglepubsubChannelBindingElement);
    base.register('googlepubsubMessageBinding', GooglepubsubMessageBindingElement);
    base.register('googlepubsubOperationBinding', GooglepubsubOperationBindingElement);
    base.register('googlepubsubServerBinding', GooglepubsubServerBindingElement);
    // HTTP
    base.register('httpChannelBinding', HttpChannelBindingElement);
    base.register('httpMessageBinding', HttpMessageBindingElement);
    base.register('httpOperationBinding', HttpOperationBindingElement);
    base.register('httpServerBinding', HttpServerBindingElement);
    // IBM MQ
    base.register('ibmmqChannelBinding', IbmmqChannelBindingElement);
    base.register('ibmmqMessageBinding', IbmmqMessageBindingElement);
    base.register('ibmmqOperationBinding', IbmmqOperationBindingElement);
    base.register('ibmmqServerBinding', IbmmqServerBindingElement);
    // JMS
    base.register('jmsChannelBinding', JmsChannelBindingElement);
    base.register('jmsMessageBinding', JmsMessageBindingElement);
    base.register('jmsOperationBinding', JmsOperationBindingElement);
    base.register('jmsServerBinding', JmsServerBindingElement);
    // Kafka
    base.register('kafkaChannelBinding', KafkaChannelBindingElement);
    base.register('kafkaMessageBinding', KafkaMessageBindingElement);
    base.register('kafkaOperationBinding', KafkaOperationBindingElement);
    base.register('kafkaServerBinding', KafkaServerBindingElement);
    // Mercure
    base.register('mercureChannelBinding', MercureChannelBindingElement);
    base.register('mercureMessageBinding', MercureMessageBindingElement);
    base.register('mercureOperationBinding', MercureOperationBindingElement);
    base.register('mercureServerBinding', MercureServerBindingElement);
    // MQTT
    base.register('mqttChannelBinding', MqttChannelBindingElement);
    base.register('mqttMessageBinding', MqttMessageBindingElement);
    base.register('mqttOperationBinding', MqttOperationBindingElement);
    base.register('mqttServerBinding', MqttServerBindingElement);
    // MQTT 5
    base.register('mqtt5ChannelBinding', Mqtt5ChannelBindingElement);
    base.register('mqtt5MessageBinding', Mqtt5MessageBindingElement);
    base.register('mqtt5OperationBinding', Mqtt5OperationBindingElement);
    base.register('mqtt5ServerBinding', Mqtt5ServerBindingElement);
    // NATS
    base.register('natsChannelBinding', NatsChannelBindingElement);
    base.register('natsMessageBinding', NatsMessageBindingElement);
    base.register('natsOperationBinding', NatsOperationBindingElement);
    base.register('natsServerBinding', NatsServerBindingElement);
    // Redis
    base.register('redisChannelBinding', RedisChannelBindingElement);
    base.register('redisMessageBinding', RedisMessageBindingElement);
    base.register('redisOperationBinding', RedisOperationBindingElement);
    base.register('redisServerBinding', RedisServerBindingElement);
    // SNS
    base.register('snsChannelBinding', SnsChannelBindingElement);
    base.register('snsMessageBinding', SnsMessageBindingElement);
    base.register('snsOperationBinding', SnsOperationBindingElement);
    base.register('snsServerBinding', SnsServerBindingElement);
    // Solace
    base.register('solaceChannelBinding', SolaceChannelBindingElement);
    base.register('solaceMessageBinding', SolaceMessageBindingElement);
    base.register('solaceOperationBinding', SolaceOperationBindingElement);
    base.register('solaceServerBinding', SolaceServerBindingElement);
    // SQS
    base.register('sqsChannelBinding', SqsChannelBindingElement);
    base.register('sqsMessageBinding', SqsMessageBindingElement);
    base.register('sqsOperationBinding', SqsOperationBindingElement);
    base.register('sqsServerBinding', SqsServerBindingElement);
    // STOMP
    base.register('stompChannelBinding', StompChannelBindingElement);
    base.register('stompMessageBinding', StompMessageBindingElement);
    base.register('stompOperationBinding', StompOperationBindingElement);
    base.register('stompServerBinding', StompServerBindingElement);
    // WebSocket
    base.register('webSocketChannelBinding', WebSocketChannelBindingElement);
    base.register('webSocketMessageBinding', WebSocketMessageBindingElement);
    base.register('webSocketOperationBinding', WebSocketOperationBindingElement);
    base.register('webSocketServerBinding', WebSocketServerBindingElement);

    return base;
  },
};

export default asyncApi2;
