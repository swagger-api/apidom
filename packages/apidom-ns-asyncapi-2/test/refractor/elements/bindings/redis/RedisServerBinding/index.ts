import { expect } from 'chai';
import { sexprs } from 'apidom';

import { RedisServerBindingElement } from '../../../../../../src';

describe('refractor', function () {
  context('elements', function () {
    context('RedisServerBindingElement', function () {
      specify('should refract to semantic ApiDOM tree', function () {
        const redisServerBindingElement = RedisServerBindingElement.refract({});

        expect(sexprs(redisServerBindingElement)).toMatchSnapshot();
      });
    });
  });
});
