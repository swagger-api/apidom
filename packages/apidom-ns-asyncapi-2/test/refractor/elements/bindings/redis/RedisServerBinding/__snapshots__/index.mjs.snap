// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refractor elements RedisServerBindingElement should refract to semantic ApiDOM tree 1`] = `(RedisServerBindingElement)`;
